zstackctl/zstackctl/ctl.py

```
@@ -10308,19 +10310,89 @@ def build_catalina_opts(self, heap):
             raise CtlError('unexpected size: %s' % args.size)
 
-        heap, psize, ratio, sint = deploymentProfiles[s]
+        heap, psize, ratio, sint, max_thread, enable_elab, ping_interval, ping_parallel = deploymentProfiles[s]
         # tag::setdeploymentconfig[]
         subprocess.getstatusoutput("zstack-ctl setenv CATALINA_OPTS='%s'" % self.build_catalina_opts(heap))
         subprocess.getstatusoutput("zstack-ctl configure DbFacadeDataSource.maxPoolSize=%s" % psize)
         subprocess.getstatusoutput("zstack-ctl configure KvmHost.maxThreads.ratio=%s" % ratio)
         subprocess.getstatusoutput("zstack-ctl configure Prometheus.scrapeInterval=%s" % sint)
+        gc_success, gc_fail = [], []
+        if max_thread is not None:
+            subprocess.getstatusoutput("zstack-ctl configure ThreadFacade.maxThreadNum=%s" % max_thread)
```

Ruff (0.15.9)

[error] Starting a process with a shell, possible injection detected

(S605)